File: domain/exercises-domain/src/main/java/com/fit/fitndflow/app/domain/exercises/repository/ExercisesRepository.java
package com.fit.fitndflow.app.domain.exercises.repository;

import com.fit.fitndflow.app.domain.common.models.ExerciseModel;
import com.fit.fitndflow.app.domain.common.repository.CommonRepository;
import java.util.List;

public interface ExercisesRepository extends CommonRepository {
    List<ExerciseModel> addNewExercise(String exerciseName, String language, int categoryId) throws Exception;
    List<ExerciseModel> modifyExercise(int exerciseId, String exerciseName, String language, int categoryId) throws Exception;
    List<ExerciseModel> deleteExercise(Integer integer) throws Exception;
}

File: domain/exercises-domain/src/main/java/com/fit/fitndflow/app/domain/exercises/repository/ExercisesRepository.kt
package com.fit.fitndflow.app.domain.exercises.repository

import com.fit.fitndflow.app.domain.common.models.ExerciseModel
import com.fit.fitndflow.app.domain.common.repository.CommonRepository

interface ExercisesRepository : CommonRepository {
    suspend fun addNewExercise(
        exerciseName: String,
        language: String,
        categoryId: Int
    ): List<ExerciseModel>

    suspend fun modifyExercise(
        exerciseId: Int,
        exerciseName: String,
        language: String,
        categoryId: Int
    ): List<ExerciseModel>

    suspend fun deleteExercise(exerciseId: Int): List<ExerciseModel>
}
